bin/check-postgres-replication.rb

```diff
@@ -103,6 +103,9 @@ class CheckPostgresReplicationStatus < Sensu::Plugin::Check::CLI
   def run
     ssl_mode = config[:ssl] ? 'require' : 'prefer'
 
+    critical 'Slave host is undefined' unless config[:slave_host] && !['localhost', '127.0.0.1'].include?(config[:slave_host])
```

Comment:
If we need to chain conditionals with an `unless` lets use an `if !condition1 && condition2` as its very easy to introduce subtle bugs and makes it harder to read (the latter being an opinion). I find this easier to read and less error prone to an accidental bug being introduced by someone who is less familiar with the nuances. In this case there is a default of `localhost` defined so its not clear to me if the first part of the conditional makes sense but checking specifically for `.nil?` is more revealing to its intent IMO. I am still not quite 100% sure I understand this since your attempting to assert that its not defined when there would always be a default. Reading the options I think the assumption is that this typically runs on the slaves (hence the default of slave to `localhost` and that the master must be specified. This sounds like you want the reverse. I am wondering if the cleanest way to handle that is to basically make them both required flags to set (which is a breaking change). Thoughts?

```suggestion
    critical 'Slave host is undefined' if config[:slave_host].nil? && ['localhost', '127.0.0.1'].include?(config[:slave_host])
```

Reply:
> hence the default of slave to localhost

Perhaps there are scenarios where a PG slave is listening on localhost but ATM I can't think of one.

>  the cleanest way to handle that is to basically make them both required flag

I tend to agree. Set both master and slave and `critical` if they are localhost.

Does that sound like a good path forward @majormoses ?

Reply:
The only viable scenario I can think of is in the context of local testing or in a large docker cluster where they might be on the same host (not advisable) but with different ports but we don't currently support sending different ports so I dont think that would affect anyone with this. I think its fine to move forward with that approach.

Reply:
> we don't currently support sending different ports so I dont think that would affect anyone with this

Might be a worthy consideration for a follow up PR.

> I think its fine to move forward with that approach.

Thanks, I'll update this PR.